### PR TITLE
fix small errors, and incorrect overlap boundries

### PR DESCRIPTION
**Pull request by msettles**
addresses #117, #118, #119 Overlapping had some issues with a <= (should be < ) and when overlap was the same size as kmer. fixed and varified

### CODE REVIEW

File: hts_Overlapper/src/hts_Overlapper.h [hunk 1]
```diff
@@ -213,7 +213,7 @@ spReadBase getOverlappedReads(Read &r1, Read &r2, const seqLookup &seq1Map,  con
             }
         }
     } 
-    for (size_t bp = seq2.length() - (checkLengths + kmer); bp < seq2.length() - kmer ; ++bp) {
+    for (size_t bp = seq2.length() - (checkLengths + kmer); bp <= (seq2.length() - kmer) ; ++bp) {
```
Comment (dstreett) [inline hts_Overlapper/src/hts_Overlapper.h]: Same question as above.

File: hts_AdapterTrimmer/src/hts_AdapterTrimmer.h [hunk 2]
```diff
@@ -199,8 +199,8 @@ void getOverlappedReads(Read &r1, Read &r2, const seqLookup &seq1Map,  const dou
             }
         }
     } 
-    for (size_t bp = seq2.length() - (checkLengths + kmer); bp < seq2.length() - kmer ; ++bp) {
-        // check last checkLengths kmers at the end of the read2 (mins)
+    for (size_t bp = seq2.length() - (checkLengths + kmer); bp <= (seq2.length() - kmer) ; ++bp) {
```
Comment (dstreett) [inline hts_AdapterTrimmer/src/hts_AdapterTrimmer.h]: Why the equal <= ? Just making sure that is intentional.

File: common/src/counters.h [hunk 3]
```diff
@@ -150,24 +150,16 @@ class Counters {
 
 private:
     virtual void check_write() {
-        outStats.open(fStats, std::ios::in | std::ios::out);
+        outStats.open(fStats, std::ios::in || std::ios::out);
```
[the remark this answers is not in the document]
Comment (msettles) [reply common/src/counters.h]: fixed in commit https://github.com/ibest/HTStream/pull/121/commits/111d6698c03e13e9b5402e496250429994fe2751

File: hts_AdapterTrimmer/src/hts_AdapterTrimmer.h [hunk 4]
```diff
@@ -136,21 +136,21 @@ unsigned int checkIfOverlap(Read &r1, Read &r2, size_t loc1, size_t loc2, const
...
     std::advance(i2, loc2_t);
-    if (maxLoop <= minOverlap || !threshold_mismatches(i1, i2, maxLoop, maxMis) ) {
+    if (maxLoop < minOverlap || !threshold_mismatches(i1, i2, maxLoop, maxMis) ) {
         // no overlap identified
         return 0;
     }
     // overlap exists thats meet maxMis criteria
-    size_t read1_bp;
-    size_t read2_bp;
+    if (!noFixBases){
```
[the remark this answers is not in the document]
Comment (msettles) [reply hts_AdapterTrimmer/src/hts_AdapterTrimmer.h]: Maybe put into its own function later

**Pull request merged**